webdriver/tests/bidi/script/evaluate/csp_tentative.py

```diff
@@ -0,0 +1,92 @@
+import pytest
+import asyncio
+
+from webdriver.bidi.modules.script import ContextTarget
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "expression",
+    [
+        "2 + 1",
+        "eval('2 + 1')",
+        "new Function('return 2 + 1')()",
+        """
+          new Promise(r => {
+            setTimeout(() => {
+              r(eval('2 + 1'));
+            }, 0);
+          })
+        """,
+        """
+          (async () => {
+            await new Promise(r => setTimeout(r, 0));
+            return eval("2 + 1");
+          })()
+        """,
+        "window.inlineScriptEval()",
+        "window.preloadScriptEval()",
+        "window.document.body.onclick()",
```

Comment:
Shouldn't this fail if the CSP disallows eval? In other words, if I am testing that my CSP works, I want this to fail, correct?

Reply:
Deciding this is the goal of the discussion in https://github.com/w3c/webdriver-bidi/issues/1024

This test covers all the scenarios I could think about regarding eval/new Function usage from BiDi. At the moment Firefox fails all of them, and Chrome passes all of them except for the async use cases.

But I agree with you and I think it would be better if scripts defined in the content page were not allowed to bypass CSP, even if they are called from a WebDriver BiDi evaluated script. So both window.inlineScriptEval and window.document.body.onclick should fail IMO.